From e989af4d14a58390521ee2c74342a99b8fe60c7c Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:25:21 -0400
Subject: [alsa] hda: Make codec object as a parent for input beep devices

Message-id: <140972823573-196-patchset-jkysela@redhat.com>
Patchwork-id: 90896
O-Subject: [RHEL 7.1 PATCH 195/442] ALSA: hda - Make codec object as a parent for input beep devices
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

Instead of the controller, the new codec object is assigned as a
parent for the hd-audio beep input devices, just like already done for
PCM and hwdep.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: f2606a807935fab0d5a2806673dcc4ce4f1f40e4)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_beep.c b/sound/pci/hda/hda_beep.c
index d403981..88bb084 100644
--- a/sound/pci/hda/hda_beep.c
+++ b/sound/pci/hda/hda_beep.c
@@ -166,7 +166,7 @@ static int snd_hda_do_attach(struct hda_beep *beep)
  input_dev->evbit[0] = BIT_MASK(EV_SND);
  input_dev->sndbit[0] = BIT_MASK(SND_BELL) | BIT_MASK(SND_TONE);
  input_dev->event = snd_hda_beep_event;
- input_dev->dev.parent = codec->bus->card->dev;
+ input_dev->dev.parent = &codec->dev;
  input_set_drvdata(input_dev, beep);
 
  err = input_register_device(input_dev);
-- 
1.7.1